### PR TITLE
Feature/draggable file item

### PR DESCRIPTION
**Pull request by AndriiSameliuk**

### CODE REVIEW

**Comment by AndriiSameliuk** [comment]
@dkalinin, PTAL.

File: app/Store/Actions/index.js [hunk 1]
```diff
@@ -20,14 +18,13 @@ export default {
           store.commit('setStatus', '');
         });
   },
-  fetchFiles(store) {
+  fetchFiles(store, folderId) {
     store.commit('setStatus', 'Loading files...');
-    const promise = FileService.getFiles(store.state.currentFolder);
+    const promise = FileService.getFiles(folderId);
```
Comment (dkalinin) [inline app/Store/Actions/index.js]: Please merge service-provider to this branch.

Comment (AndriiSameliuk) [reply app/Store/Actions/index.js]: I branch service-provider from development.

File: app/View/Component/FileList/index.js [hunk 2]
```diff
@@ -154,8 +158,9 @@ export class FileList extends DraggableComponent {
   _onDrop(event) {
     event.preventDefault();
     event.stopPropagation();
-    console.log(Array.from(event.dataTransfer.files));
-    this.dispatch('addItems', Array.from(event.dataTransfer.files), this.state.currentFolder);
+    if (!event.dataTransfer.getData('fileId')) {
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Please check event.dataTransfer.files.

File: app/View/Component/FileItem/index.js [hunk 3]
```diff
@@ -190,10 +193,26 @@ export class FileItem extends DraggableComponent {
   _onDrop(event) {
     event.preventDefault();
     event.stopPropagation();
-    this.folderDropCallback(Array.from(event.dataTransfer.files), this.id);
+
+    if (event.dataTransfer.getData('fileId') !== this.id) {
+      console.log(event.dataTransfer.files);
+      if (event.dataTransfer.files.length === 0) {
+        this.moveCallback((event.dataTransfer.getData('fileId')), this.id);
```
[the remark this answers is not in the document]
Comment (dkalinin) [reply app/View/Component/FileItem/index.js]: Also check if getData(fileId) exists.

File: app/Store/Actions/index.js [hunk 4]
```diff
@@ -87,27 +88,44 @@ export default {
...
+          store.commit('setServerErrors', errorMessage);
         })
         .finally(() => {
         });
   },
 
-  setCurrentFolder(store, folderId) {
-    store.commit('setCurrentFolder', folderId);
+  setCurrentFolder(store, folderId, name) {
+    store.commit('setCurrentFolder', folderId, name);
```
Comment (dkalinin) [inline app/Store/Actions/index.js]: Let's set folder model.

File: app/View/Component/FileItem/index.js [hunk 5]
```diff
@@ -193,13 +251,13 @@ export class FileItem extends DraggableComponent {
...
     event.stopPropagation();
-
-    if (event.dataTransfer.getData('fileId') !== this.id) {
-      console.log(event.dataTransfer.files);
-      if (event.dataTransfer.files.length === 0) {
-        this.moveCallback((event.dataTransfer.getData('fileId')), this.id);
+    const fileId = event.dataTransfer.getData(FILE_ID);
+    const uploadedFiles = event.dataTransfer.files;
+    if (fileId && fileId !== this.id) {
+      if (uploadedFiles.length === 0) {
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: Drop on folder doesn't work.

File: app/Store/Actions/index.js [hunk 6]
```diff
@@ -99,9 +99,9 @@ export default {
         });
   },
 
-  setCurrentFolder(store, folderId, name) {
-    store.commit('setCurrentFolder', folderId, name);
-    store.dispatch('fetchFiles', folderId);
+  setCurrentFolder(store, file) {
```
Comment (dkalinin) [inline app/Store/Actions/index.js]: folder

File: app/Store/Actions/index.js [hunk 7]
```diff
@@ -124,8 +124,8 @@ export default {
     return promise
         .then((file) => {
```
Comment (dkalinin) [inline app/Store/Actions/index.js]: folder

File: app/Store/Mutations/index.js [hunk 8]
```diff
@@ -31,22 +31,15 @@ export default {
       name: newName,
     };
   },
-  setCurrentFolder(state, folderId, name) {
-    state.currentFolder = {
-      id: folderId,
-      name: name,
-    };
+  setCurrentFolder(state, file) {
```
Comment (dkalinin) [inline app/Store/Mutations/index.js]: folder

File: app/Store/Mutations/index.js [hunk 9]
```diff
@@ -31,22 +31,15 @@ export default {
...
   setServerErrors(state, serverError) {
     state.serverError = serverError;
   },
 
-  setMainFolder(state, folderId, name) {
-    state.mainFolderId = {
-      id: folderId,
-      name: name,
-    };
+  setMainFolder(state, file) {
```
Comment (dkalinin) [inline app/Store/Mutations/index.js]: folder

File: app/View/Component/FileItem/index.js [hunk 10]
```diff
@@ -337,7 +337,10 @@ export class FileItem extends DraggableComponent {
     setTimeout(() => {
       if (this.clickCount === 1) {
         if (this.type === 'folder') {
-          this.clickedFolderCallback(this.id, this.name);
+          this.clickedFolderCallback({
+            id: this.id,
+            name: this.name,
+          });
```
Comment (dkalinin) [inline app/View/Component/FileItem/index.js]: Let's have file model as a param in constructor and as a field in FileItem.

File: app/View/Component/FileList/index.js [hunk 11]
```diff
@@ -51,9 +51,7 @@ export class FileList extends DraggableComponent {
 
     moveToRootFolder.onClick(() => {
       console.log('fdfsfsfs');
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Please remove.

File: app/View/Component/FileList/index.js [hunk 12]
```diff
@@ -51,9 +51,7 @@ export class FileList extends DraggableComponent {
 
     moveToRootFolder.onClick(() => {
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: Go to root folder.

File: app/View/Component/FileList/index.js [hunk 13]
```diff
@@ -126,8 +124,8 @@ export class FileList extends DraggableComponent {
         this.dispatch('moveFile', fileId, parentFolderId);
       });
 
-      fileItem.onFolderClick((folderId, name) => {
-        this.dispatch('setCurrentFolder', folderId, name);
+      fileItem.onFolderClick((file) => {
```
Comment (dkalinin) [inline app/View/Component/FileList/index.js]: folder

**Comment by AndriiSameliuk** [comment]
@dkalinin, fixed, PTAL.

**Pull request merged**